packages/livekit-rtc/src/utils.ts

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2024 LiveKit, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+import EventEmitter, { once } from 'events';
+
+/** @internal */
+export class Queue<T> {
```

Comment:
Seems like the Queue isn't needed, just Promise/Future?

Reply:
this promise is reused more than one time so it's more ergonomic to use queue for this